onnxruntime/core/providers/xnnpack/nn/resize.cc

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/xnnpack/nn/resize.h"
+
+#include <utility>
+
+#include "core/common/common.h"
+#include "core/framework/op_kernel.h"
+#include "core/optimizer/initializer.h"
+
+namespace onnxruntime {
+namespace xnnpack {
+
+bool Resize::IsOnnxNodeSupported(const NodeUnit& node_unit,
+                                 const GraphViewer&) {
+  bool supported = false;
+  do {
+    // Resize has 1-4 input.
+    const auto& inputs = node_unit.Inputs();
+    const auto& x_arg = inputs[0].node_arg;
+
+    const auto* x_type = x_arg.TypeAsProto();
+    if (x_type == nullptr ||
+        (x_type->tensor_type().elem_type() != ONNX_NAMESPACE::TensorProto_DataType_FLOAT &&
+         x_type->tensor_type().elem_type() != ONNX_NAMESPACE::TensorProto_DataType_UINT8 &&
+         x_type->tensor_type().elem_type() != ONNX_NAMESPACE::TensorProto_DataType_INT8)) {
+      break;
+    }
+    const auto* x_shape = x_arg.Shape();
+    //'bilinear' == 2-D input or 4-D input with outermost 2 scales as 1 or
+    // 4-D input with outermost and innermost scales as 1
+    // but we just support 4-d tensor for now
```

Comment:
nit: this is also checking the channels value (dim(1)) is known isn't it, which is a hard requirement if we create the xnnpack kernel in the ctor. If that's the case, the comment should call that out.

Reply:
Yeah,  we don't support dynamic shape except Batchsize.
High-lighed in comments